### ISSUE TITLE
Cannot read property 'slice' of undefined

### ISSUE THREAD

**Issue by srakshit** [joshcarr/react-structured-filter#7]
Hi @joshcarr ,

I'm using the minimalist feature of the filter like

``` javascript
<StructuredFilter
    placeholder="Search..."
    options={[
        {category:"Name",type:"text"},
        {category:"Price",type:"number"},
    ]}
/>
```

and I'm getting the error 

``` javascript
Uncaught TypeError: Cannot read property 'slice' of undefined
getStateFromProps 
Tokenizer 
constructComponentWithoutOwner 
```

Could you please help.

Thanks
Subham

**Comment by joshcarr** [comment]
You must either provide a `value` or a `defaultValue` prop. Unfortunately, the docs don't reflect that. But if you look at them you can see how to use one or the other of those props to make the component controlled or uncontrolled: 

https://github.com/joshcarr/react-structured-filter/blob/master/docs/index.md

**Comment by joshcarr** [comment]
Fixed this in d484849 and released v4.3.0

**Comment by srakshit** [comment]
@joshcarr Thanks! It works :)
